cwltest/utils.py

```diff
@@ -73,6 +76,21 @@ def compare_location(expected, actual):
         return
     if actual.get("class") == "Directory":
         actual[comp] = actual[comp].rstrip("/")
+        exist_fun = os.path.isdir  # type: Callable[[str], bool]
+    else:
+        exist_fun = os.path.isfile
+
+    if "path" in actual:
+        path = urllib.parse.urlparse(actual["path"]).path
+    else:
+        path = urllib.parse.urlparse(actual["location"]).path
```

Comment:
And another test involving `location`?

Reply:
I added a test for `Directory` with `location`.